### PR TITLE
Check for downgrade and non-trial subscriptions

### PR DESCRIPTION
**Pull request by gbova**
Check for downgrade and non-trial subscriptions _before_ sending the downgrading email.

`billing_account_info_form.is_downgrade()` is more accurate than `ConfirmSelectedPlanView.is_upgrade` because `self.current_subscription` is updated to the new subscription before the user reaches the ConfirmSelectedPlanView.

### CODE REVIEW

File: corehq/apps/domain/forms.py [hunk 1]
```diff
@@ -1696,6 +1696,8 @@ def save(self, commit=True):
     def is_downgrade(self):
```
Comment (nickpell) [inline corehq/apps/domain/forms.py]: This function is a bit of a misnomer now.  Might be better to rename to ```is_downgrade_from_paid_plan``` or something like that.

Comment (nickpell) [reply corehq/apps/domain/forms.py]: (Fine to address in a separate PR)

**Review by nickpell — APPROVED** [review]

File: corehq/apps/domain/forms.py (hunk 1 above)
Comment (gbova) [reply corehq/apps/domain/forms.py]: You're right, this makes more sense

File: corehq/apps/domain/forms.py [hunk 2]
```diff
@@ -1693,9 +1693,11 @@ def save(self, commit=True):
             )
             return False
 
-    def is_downgrade(self):
+    def is_downgrade_from_paid_plan(self):
```
[the remark this answers is not in the document]
Comment (gbova) [reply corehq/apps/domain/forms.py]: It _definitely_ does, thanks for stopping me

**Review by nickpell — APPROVED** [review]

**Pull request merged**